shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -193,12 +219,30 @@
   overlay_view.frame = flutter_view_.get().bounds;
   overlay_view.autoresizingMask =
       (UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight);
-  std::unique_ptr<IOSSurface> ios_surface = overlay_view.createSurface;
+  std::unique_ptr<IOSSurface> ios_surface = overlay_view.createSoftwareSurface;
   std::unique_ptr<Surface> surface = ios_surface->CreateGPUSurface();
   overlays_[overlay_id] = std::make_unique<FlutterPlatformViewLayer>(
       overlay_view, std::move(ios_surface), std::move(surface));
 }
 
+void FlutterPlatformViewsController::EnsureGLOverlayInitialized(
+    int64_t overlay_id,
+    std::shared_ptr<IOSGLContext> gl_context,
+    GrContext* gr_context) {
+  if (overlays_.count(overlay_id) != 0) {
+    return;
+  }
+  FlutterOverlayView* overlay_view = [[FlutterOverlayView alloc] init];
+  overlay_view.frame = flutter_view_.get().bounds;
+  overlay_view.autoresizingMask =
+      (UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight);
+  std::unique_ptr<IOSSurfaceGL> ios_surface =
+      [overlay_view createGLSurfaceWithContext:std::move(gl_context)];
+  std::unique_ptr<Surface> surface = ios_surface->CreateSecondaryGPUSurface(gr_context);
+  overlays_[overlay_id] = std::make_unique<FlutterPlatformViewLayer>(
+      overlay_view, std::move(ios_surface), std::move(surface));
```

Comment:
Nit: It is hard to tell the ownership model of overlay_view from reading this bit of code. Can you make the constructor take a scoped_nsobject instead please?

Reply:
done